File: lua_helper/policy_helper.lua
ProductPolicy = {}

function ProductPolicy:Load(key, item)
  App:Call('ProductPolicy::Load', key, item)
end

function ProductPolicy:Get(name)
  App:Call('ProductPolicy::Get', name)
end

function ProductPolicy:Set(name, value)
  App:Call('ProductPolicy::Set', name, value)
end

function ProductPolicy:Save(name, value)
  if name ~= nil then
    App:Call('ProductPolicy::Set', name, value)
  end
  App:Call('ProductPolicy::Save')
end

File: luaengine/PolicyHelper.cpp
#include "precomp.h"
#include "../utility/ProductPolicyEditor.h"

#include "LuaEngine.h"

EXTERN_C {
    int lua_productpolicy_call(lua_State* L, const char *funcname, int top, int base) {
        int ret = 0;
        Token v = { TOK_UNSET };

        std::string func = funcname;
        if (func == "productpolicy::load") {
            ProductPolicyLoad(lua_tostring(L, base + 2), lua_tostring(L, base + 3));
        } else if (func == "productpolicy::get") {
            v.str = s2w(lua_tostring(L, base + 2));
            ProductPolicyGet(v.str.c_str());
        } else if (func == "productpolicy::set") {
            v.str = s2w(lua_tostring(L, base + 2));
            v.iVal = lua_tointeger(L, base + 3);
            ProductPolicySet(v.str.c_str(), v.iVal);
        } else if (func == "productpolicy::save") {
            ProductPolicySave();
        }
        return ret;
    }
}
